### PR TITLE
Create infrastructure to run google test cases

### PR DESCRIPTION
**Pull request by branupama**

### CODE REVIEW

File: test/utest-utils.cpp [hunk 1]
```diff
@@ -0,0 +1,28 @@
+#include "types.hpp"
```
Comment (SunnySrivastava1984) [inline test/utest-utils.cpp:1]: Please remove unnecessary header files.

**Review by SunnySrivastava1984 — CHANGES_REQUESTED** [review]

**Comment by geissonator** [comment]
What is this "P11_Dev" branch? Why aren't we starting upstream with this?

File: test/utest-utils.cpp (hunk 1 above)
Comment (branupama) [reply test/utest-utils.cpp]: Done.

**Comment by branupama** [comment]
> What is this "P11_Dev" branch? Why aren't we starting upstream with this?

P11_Dev is new architecture design for VPD. We just started building this repo.

**Comment by SunnySrivastava1984** [comment]
> What is this "P11_Dev" branch? Why aren't we starting upstream with this?
This is a branch where we are trying out some architectural changes related to VPD code. Once we are confident from our end, we will take the changes upstream.
For the time being, current upstream code is what we are pointing to, in our recipe.

**Review by SunnySrivastava1984 — APPROVED** [review]

**Comment by PriyangaRamasamy** [comment]
jenkins run tests please

**Review by GiridhariKrishnan — APPROVED** [review]

**Comment by branupama** [comment]
jenkins run tests please

File: test/meson.build [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        gmock_dep = gtest_proj.dependency('gmock')
+  else
+        assert(
+            not get_option('tests').allowed(),
+            'Googletest is required if tests are enabled'
+        )
+  endif
+endif
+
+dependecy_list = [gtest_dep, gmock_dep, sdbusplus]
```
Comment (PriyangaRamasamy) [inline test/meson.build:28]: 1. spellcheck: dependency*
2. I don't think we need sdbusplus dependency for test cases. Can you remove it and see if you get any compile time errors

File: test/utest_utils.cpp [hunk 3]
```diff
@@ -0,0 +1,24 @@
+#include "utils.hpp"
+
+#include <cassert>
+#include <iostream>
```
Comment (PriyangaRamasamy) [inline test/utest_utils.cpp:4]: Do we need iostream here?

**Comment by PriyangaRamasamy** [comment]
jenkins run tests please

File: test/meson.build (hunk 2 above)
Comment (branupama) [reply test/meson.build]: 1. Updated the spellcheck 
2. sdbusplus dependency coming from utils.cpp file

File: test/utest_utils.cpp (hunk 3 above)
Comment (branupama) [reply test/utest_utils.cpp]: removed iostream include.

File: meson.build [hunk 4]
```diff
@@ -23,37 +23,11 @@ endif
 # Setup googletest before we import any projects that also depend on it to make
```
Comment (jinuthomas) [inline meson.build:23]: move this comment also to test

File: test/meson.build [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+configuration_inc = include_directories('..', '../include')
+
+test_sources = [
+  '../src/utils.cpp',
+  '../src/logger.cpp'
+]
+
+tests = [
+  'utest_utils.cpp'
```
Comment (jinuthomas) [inline test/meson.build:38]: why file name is utest_

**Review by jinuthomas — COMMENTED** [review]
I am not sure if the names are good 
example 
TEST(UtilsTest, TestValidValue)
where did you get the variables UtilsTest and TestValidValue

**Pull request merged**